packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -95,6 +98,45 @@ exports.sourceNodes = async (
     pluginConfig,
   })
 
+  // compare data between currentSyncData and previousSyncData and update it
+  previousSyncData.entries.filter(entry =>
```

Comment:
It looks like the assignment is missing here - `filter` does not change original array. Same goes for assets array below
```suggestion
  previousSyncData.entries = previousSyncData.entries.filter(entry =>
```